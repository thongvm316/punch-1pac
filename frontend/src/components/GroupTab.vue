<template>
  <ul class="tab mt-4">
    <router-link tag="li"
      class="tab-item"
      :to="`/groups/${groupId}/users`">
      <a href="#">{{ $t('group.users') }}</a>
    </router-link>
    <router-link tag="li"
      class="tab-item"
      :to="`/groups/${groupId}/attendances`">
      <a href="#">{{ $t('group.attendances') }}</a>
    </router-link>
    <router-link tag="li"
      class="tab-item"
      :to="`/groups/${groupId}/requests`">
      <a href="#">{{ $t('group.requests') }}</a>
    </router-link>
    <router-link tag="li"
      class="tab-item"
      :to="`/groups/${groupId}/report`" exact>
      <a href="#">{{ $t('group.report') }}</a>
    </router-link>
  </ul>
</template>

<script>
export default {
  name: 'group-tab',
  props: {
    groupId: {
      type: String,
      required: true
    }
  }
}
</script>
