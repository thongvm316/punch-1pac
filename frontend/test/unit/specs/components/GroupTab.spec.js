import { shallowMount } from '@vue/test-utils'
import wrapperOps from '../../supports/wrapper'
import GroupTab from '@/components/GroupTab'

const propsData = {
  groupId: '1',
  hasUser: true
}

const localWrapperOps = {
  ...wrapperOps,
  propsData
}

describe('GroupTab.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(GroupTab, localWrapperOps)
  })

  afterEach(() => { wrapper.vm.$destroy() })

  describe('when GroupTab was mounted', () => {
    it('should render correctly', () => {
      expect(wrapper.exists()).toBeTruthy()
      expect(wrapper.isVueInstance()).toBeTruthy()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
