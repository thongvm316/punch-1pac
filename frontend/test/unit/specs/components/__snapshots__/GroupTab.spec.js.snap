// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupTab.vue when GroupTab was mounted should render correctly 1`] = `
<ul class="tab mt-4">
  <router-link-stub to="/groups/1/users" tag="li" exact="true" event="click" class="tab-item"><a href="#">Users</a></router-link-stub>
  <router-link-stub to="/groups/1/attendances" tag="li" event="click" class="tab-item"><a href="#">Attendances</a></router-link-stub>
  <router-link-stub to="/groups/1/requests" tag="li" event="click" class="tab-item"><a href="#">Requests</a></router-link-stub>
  <router-link-stub to="/groups/1/report" tag="li" event="click" class="tab-item"><a href="#">Report</a></router-link-stub>
</ul>
`;
